lib/src/observable.dart

```diff
@@ -1,85 +1,138 @@
-
-  // sync:true -> ObjectBoxException: 10001 TX is not active anymore: #101
-  static final controller = StreamController<int>.broadcast();
-
-  // The user_data is used to pass the store ptr address
-  // in case there is no consensus on the entity id between stores
-  static void _anyCallback(
-      Pointer<Void> user_data, Pointer<Uint32> mutated_ids, int mutated_count) {
-    final storeAddress = user_data.address;
-    // call schema's callback
-    final storeCallbacks = _any[storeAddress];
-    if (storeCallbacks != null) {
-      for (var i = 0; i < mutated_count; i++) {
-        storeCallbacks[mutated_ids[i]]
-            ?.call(user_data, mutated_ids, mutated_count);
-      }
-    }
+
+/// Simple wrapper used below in ObservableStore to reduce code duplication.
+/// Contains shared code for single-entity observer and the generic/global one.
+class _Observer<StreamValueType> {
+  StreamController<StreamValueType> /*?*/ controller;
+  Pointer<OBX_observer> /*?*/ _cObserver;
+  ReceivePort /*?*/ receivePort;
+
+  int get nativePort => receivePort /*!*/ .sendPort.nativePort;
+
+  set cObserver(Pointer<OBX_observer> value) {
+    _cObserver = checkObxPtr(value, 'observer initialization failed');
+    _debugLog('started');
   }
 
-  static void subscribe(Store store) {
-    syncOrObserversExclusive.mark(store);
+  Stream<StreamValueType> get stream => controller /*!*/ .stream;
 
-    final callback = Pointer.fromFunction<obx_observer>(_anyCallback);
-    final storePtr = store.ptr;
-    _anyObserver[storePtr.address] =
-        bindings.obx_observe(storePtr, callback, storePtr.cast<Void>());
-    StoreCloseObserver.addListener(store, _anyObserver[storePtr.address], () {
-      unsubscribe(store);
-    });
+  _Observer() {
+    initializeDartAPI();
+  }
+
+  // start() is called whenever user starts listen()-ing to the stream
+  void init(void Function() start) {
+    controller = StreamController<StreamValueType>(
+        onListen: start, onPause: stop, onResume: start, onCancel: stop);
   }
 
-  // #53 ffi:Pointer finalizer
-  static void unsubscribe(Store store) {
-    final storeAddress = store.ptr.address;
-    if (!_anyObserver.containsKey(storeAddress)) {
-      return;
+  // stop() is called when the stream subscription is paused or canceled
+  void stop() {
+    _debugLog('stopped');
+    if (_cObserver != null) {
+      checkObx(bindings.obx_observer_close(_cObserver));
+      _cObserver = null;
+    }
+
+    if (receivePort != null) {
+      receivePort.close();
+      receivePort = null;
     }
-    StoreCloseObserver.removeListener(store, _anyObserver[storeAddress]);
-    bindings.obx_observer_close(_anyObserver[storeAddress]);
-    _anyObserver.remove(storeAddress);
-    syncOrObserversExclusive.unmark(store);
   }
 
-  static bool isSubscribed(Store store) =>
-      _Observable._anyObserver.containsKey(store.ptr.address);
+  void _debugLog(String message) {
+    // print('Observer=${_cObserver?.address} $message');
```

Comment:
Is this a left-over from testing? This is still a draft so maybe don't care...

Reply:
On one hand, yes, on the other, it's more like a future-proofing the code for when you need to check something manually while developing :)